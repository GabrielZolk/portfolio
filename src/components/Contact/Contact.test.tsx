import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import ContactForm from './Contact';

describe('ContactForm', () => {
  test('renders form fields and buttons', () => {
    const container = render(<ContactForm />);
    expect(container).toMatchSnapshot();
  });

  test('updates form data when fields are changed', () => {
    const { getByLabelText, container } = render(<ContactForm />);

    fireEvent.change(getByLabelText(/nome/i), {
      target: { value: 'John Due' },
    });
    fireEvent.change(getByLabelText(/assunto/i), {
      target: { value: 'Teste de formulário' },
    });
    fireEvent.change(getByLabelText(/email/i), {
      target: { value: 'john@example.com' },
    });
    fireEvent.change(getByLabelText(/mensagem/i), {
      target: { value: 'Olá, gostaria de fazer um teste de formulário.' },
    });

    expect(container).toMatchSnapshot();
  });
});
