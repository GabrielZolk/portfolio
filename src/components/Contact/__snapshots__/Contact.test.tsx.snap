// Vitest Snapshot v1

exports[`ContactForm > renders form fields and buttons 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-bcXHqe fmlIsJ"
      >
        <label>
          Nome*
          <input
            name="name"
            required=""
            type="text"
            value=""
          />
        </label>
        <br />
        <label>
          Assunto
          <input
            name="subject"
            type="text"
            value=""
          />
        </label>
        <br />
        <label>
          Email*
          <input
            name="email"
            required=""
            type="email"
            value=""
          />
        </label>
        <br />
        <label>
          Mensagem*
          <textarea
            name="message"
          />
        </label>
        <br />
        <input
          class="submit"
          type="submit"
          value="Enviar"
        />
        <input
          class="submit"
          type="reset"
          value="Redefinir"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-bcXHqe fmlIsJ"
    >
      <label>
        Nome*
        <input
          name="name"
          required=""
          type="text"
          value=""
        />
      </label>
      <br />
      <label>
        Assunto
        <input
          name="subject"
          type="text"
          value=""
        />
      </label>
      <br />
      <label>
        Email*
        <input
          name="email"
          required=""
          type="email"
          value=""
        />
      </label>
      <br />
      <label>
        Mensagem*
        <textarea
          name="message"
        />
      </label>
      <br />
      <input
        class="submit"
        type="submit"
        value="Enviar"
      />
      <input
        class="submit"
        type="reset"
        value="Redefinir"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ContactForm > updates form data when fields are changed 1`] = `
<div>
  <form
    class="sc-bcXHqe fmlIsJ"
  >
    <label>
      Nome*
      <input
        name="name"
        required=""
        type="text"
        value="John Due"
      />
    </label>
    <br />
    <label>
      Assunto
      <input
        name="subject"
        type="text"
        value="Teste de formulário"
      />
    </label>
    <br />
    <label>
      Email*
      <input
        name="email"
        required=""
        type="email"
        value="john@example.com"
      />
    </label>
    <br />
    <label>
      Mensagem*
      <textarea
        name="message"
      >
        Olá, gostaria de fazer um teste de formulário.
      </textarea>
    </label>
    <br />
    <input
      class="submit"
      type="submit"
      value="Enviar"
    />
    <input
      class="submit"
      type="reset"
      value="Redefinir"
    />
  </form>
</div>
`;
